Add MoveIt Configuration Tutorial

sea-bass opened the pull request:
### Description

This PR adds a new tutorial introducing MoveIt configuration packages and the `MoveItConfigsBuilder`. This should hopefully help newcomers make sense of a little more of the overwhelming information and complexity in MoveIt.

### Checklist
- [x] **Required by CI**: Code is auto formatted using [clang-format](http://moveit.ros.org/documentation/contributing/code)
- [ ] While waiting for someone to review your request, please consider reviewing [another open pull request](https://github.com/ros-planning/moveit2/pulls) to support the maintainers

[//]: # "You can expect a response from a maintainer within 7 days. If you haven't heard anything by then, feel free to ping the thread. Thank you!"

sjahr reviewed (approved):
Very nice and helpful tutorial, thanks! I wouldn't dump it in the example section. What do you think about putting it into the How-To section and naming it "How to configure MoveIt"

sea-bass commented:
> Very nice and helpful tutorial, thanks! I wouldn't dump it in the example section. What do you think about putting it into the How-To section and naming it "How to configure MoveIt"

I agree with that, as this isn't really an example, but all the other configuration "examples" are also under this section, so I was trying to be consistent.

![image](https://user-images.githubusercontent.com/4603398/233633875-4b073627-91f1-41a0-82e3-79aa6e67f17e.png)

What do you think?

stephanie-eng commented:
Examples has this header:
![image](https://user-images.githubusercontent.com/10466537/233643735-249e76e2-05bc-4f1c-a146-7fc123d1e349.png)

I don't think that applies to this work.

stephanie-eng commented:
Some more context: at one point, we restructured the tutorials ([here](https://github.com/ros-planning/moveit2_tutorials/pull/251)) in effort to clean up content and identify outdated content. Examples became the catch-all bucket for outdated/unrefined tutorials. In fact, a quick review of the tutorials in examples shows:
| Tutorial                                 | Status                                    |
|------------------------------------------|-------------------------------------------|
| Kinematics Configuration                 | Mostly agnostic, but links to ROS 1       |
| Custom Constraint Samplers               | Stub, no content                          |
| OMPL Planner                             | Mostly agnostic, but links to ROS 1       |
| CHOMP Planner                            | ROS 1, but Mike has a PR for porting this |
| STOMP Planner                            | ROS 1                                     |
| TrajOpt Planner                          | ROS 1                                     |
| Pilz Industrial Motion Planner           | ROS 2, perhaps this should move?          |
| Sequence of multiple segments            | Part of the above tutorial?               |
| Planning Adapter Tutorials               | ROS 1                                     |
| Warehouse - Persistent Scenes and States | ROS 2                                     |

Your work seems both up to date and fits in a s a "How To", so I don't think putting it in the examples bin with all the outdated content is a good fit.

stephanie-eng commented on doc/examples/moveit_configuration/moveit_configuration_tutorial.rst:
```diff
@@ -0,0 +1,183 @@
+The recommended way of configuring MoveIt for your robot is by creating a colcon package containing the MoveIt Configuration.
+
+Suppose you would like to create a configuration package for some robot named ``my_robot``.
+To do this, you can create a Colcon package named ``my_robot_moveit_config``, whose structure is as follows:
```
```suggestion
To do this, you can create a colcon package named ``my_robot_moveit_config``, whose structure is as follows:
```

![image](https://user-images.githubusercontent.com/10466537/233646938-db534366-5f8f-454d-ba0a-fd89d0f6d70b.png)
Looks like lower case always is the "proper" way to spell colcon

sea-bass commented on doc/examples/moveit_configuration/moveit_configuration_tutorial.rst:
```diff
@@ -0,0 +1,183 @@
+
+The default location of this file is in ``config/sensors_3d.yaml``.
+
+Loading Configuration Parameters into Launch Files
```
Good call! Somehow I misinterpreted and thought these options were also readable directly from YAML, but that's not the case. Let me know how these fixes look.

sea-bass commented:
@stephanie-eng @sjahr I moved this new page, plus the Pilz and Warehouse examples to how-to guides.

I also fixed the bug that was causing "Sequence of multiple segments" to appear in the TOC. Bad heading levels.

stephanie-eng commented on doc/how_to_guides/moveit_configuration/moveit_configuration_tutorial.rst:
```diff
@@ -0,0 +1,185 @@
+            ...
+        ...
+
+By default, all parameter files that match this ``config/_*planning.yaml`` pattern will be loaded.
```
```suggestion
By default, all parameter files that match this ``config/*_planning.yaml`` pattern will be loaded.
```

stephanie-eng reviewed (approved):
Other than the one small change above, this looks great :heart: 

Thank you for writing this!
